website/content/api-docs/system/config-ui-custom-messages.mdx

```diff
@@ -0,0 +1,242 @@
+
+- `authenticated` `(bool: <optional>)` - A flag indicating whether to only include custom messages whose authenticated property is set to the value specified in the results. If this parameter is omitted, the results include both authenticated and unauthenticated custom messages.
+
+- `type` `(string: <optional>)` - A flag that can only be set to either `banner` or `modal`. It indicates whether to only include custom messages whose type property is set to the value specified in the results. If this parameter is omitted, the results include both types of custom messages.
+
+### Sample request
+
+```shell-session
+$ curl --header "X-Vault-Token: ..." \
+    --request LIST \
+    http://127.0.0.1:8200/v1/sys/config/ui/custom-messages?active=true&authenticated=false&type=modal
+```
+
+### Sample response
+
+```json
+{
+  "data": {
+    "key_info": {
+      "01234567-89ab-cdef-0123-456789abcdef": {
+        "title": "Post-login Advisory",
+        "type": "modal",
+        "authenticated": false,
+        "start_time": "2024-01-01T00:00:00.000000000Z",
+        "end_time": null
+      }
+    },
+    "keys": [
+      "01234567-89ab-cdef-0123-456789abcdef"
+    ]
+  }
+}
+```
+
+## Create Custom Message
+
+This endpoint allows the creation of a new custom message. Vault allows up to 100 custom messages
+per namespace. If this endpoint is called to create a custom message when the maximum number of
+custom messages already exists, an error will be returned.
+
+| Method | Path                              |
+| :----- | :-------------------------------- |
+| `POST` | `/sys/config/ui/custom-messages/` |
+
+### Parameters
+
+- `title` `(string: <required>)` - The title of the custom message.
+
+- `authenticated` `(bool: <optional>)` - A flag indicating whether the custom message is displayed post-login (`true`) or pre-login (`false`). If this parameter is omitted, the custom message is created as post-login.
+
+- `type` `(string: <optional>)` - The type of the custom message. Must be either `banner` or `modal`. If this parameter is omitted, the custom message type is set to `banner`.
+
+- `message` `(string: <required>)` - The base64 encoded message body of the custom message.
+
+- `link` `(Map: <optional>)` - A link object that contains a single property. The property key is the link title and its value is the URL.
+
+- `options` `(Map: <optional>)` - A map of custom UI properties to set for the custom message.
+
+- `start_time` `(string: <required>)` - A RFC3339 formatted timestamp that marks the beginning of the custom message's active period.
+
+- `end_time` `(string: <optional>)` - A RFC3339 formatted timestamp that marks the end of the custom message's active period. If no end time is provided, the custom message's active period never ends.
+
+### Sample payload
+
+```json
+{
+  "title": "Post-login Advisory",
+  "authenticated": false,
```

Comment:
If authenticated is `false` here, isn't this a pre-login advisory?

Reply:
Thanks for spotting this. I've pushed a correction.